ui/static/js/api.js

```diff
@@ -0,0 +1,52 @@
+
+var regexpMetric = /([a-zA-Z_:][a-zA-Z0-9_:]*)(?:{(.*)})?\s+((?:\d*\.)?\d+(?:e\d+)?)(?:\s+(\d+))?/
+var regexpLabels = /([a-zA-Z_][a-zA-Z0-9_]*)\s?=\s?"([^"\\\\]*(?:\\.[^"\\\\]*)*)"/gm
+
+function parseLabels(metricName, rawLabelsStr) {
+    var labels = {"__name__": metricName};
+    if (rawLabelsStr != undefined) {
+        while(match = regexpLabels.exec(rawLabelsStr)){
+            labels[match[1]] = match[2];
+        }
+    }
+    return labels;
+}
+
+function parseSample(txtLine, defaultTimestampS) {
+    var match = regexpMetric.exec(txtLine);
+    if (match != null) {
+        let labels = parseLabels(match[1], match[2]);
+        let ts = parseInt(match[4]) || defaultTimestampS;
+        return {"metric": labels, "value": [ts, match[3]]};
+    }
+    return null;
+}
+
+function handleSimulationResult(result) {
+    let jsonres = JSON.parse(result);
+    let writers = jsonres.writers
+
+    var html = "<dl>";
+    $.each(writers, function( name, writerMsg ) {
```

Comment:
OPT: The way your arguments are set in function calls is not consistent

Reply:
Done